lemur/plugins/lemur_acme/nsone.py

```diff
@@ -0,0 +1,360 @@
+"""ACME DNS providor for NS1"""
+import json
+import inspect
+import time
+import requests
+
+import lemur.common.utils as utils
+import lemur.dns_providers.util as dnsutil
+from lemur.extensions import metrics, sentry
+from flask import current_app
+
+REQUIRED_VARIABLES = [
+    "ACME_NSONE_KEY",
+]
+
+
+def get_zones():
+    """
+    Retrieve authoritative zones from the NS1 API and return a list of zones
+
+    :raise: Exception
+    :return: list of Zone Objects
+    """
+    _check_conf()
+    path = f"/v1/zones"
+    zones = []
+    function = inspect.currentframe().f_code.co_name
+    log_data = {
+        "function": function
+    }
+    try:
+        records = _get(path)
+        log_data["message"] = "Retrieved Zones Successfully"
+        current_app.logger.debug(log_data)
+    except Exception as err:
+        sentry.captureException()
+        log_data["message"] = "Failed to Retrieve Zone Data"
+        log_data["error"] = err
+        current_app.logger.debug(log_data)
+        raise
+    for record in records:
+        zone = record['zone']
+        if record['primary']['enabled']:
```

Comment:
For my use case, I need attestation for zones that are not enabled. Removing this line worked for me.